File: iOS/DateMate/UI/Detail/SubFirstViewModel.swift
//
//  SubFirstViewModel.swift
//  DateMate
//
//  Created by 홍희표 on 2025/11/13.
//

import Foundation
import SwiftUI
import Combine

class SubFirstViewModel: ObservableObject {
    private let userRepository: UserRepository
    
    @Published var data: String
    
    private var cancellables = Set<AnyCancellable>()
    
    func signOut() {
        userRepository.getSignOutResultStream()
            .receive(on: DispatchQueue.main)
            .sink { result in
                switch result.state {
                case .Loading: break
                case .Success:
                    self.userRepository.storeUserProfile(nil)
                case .Error: break
                }
            }
            .store(in: &cancellables)
    }
    
    init(_ userRepository: UserRepository, data: String) {
        self.userRepository = userRepository
        self.data = data
    }
}

File: Android/app/src/main/java/com/hhp227/datemate/ui/detail/SubFirstViewModel.kt
package com.hhp227.datemate.ui.detail

import androidx.lifecycle.SavedStateHandle
import androidx.lifecycle.ViewModel
import androidx.lifecycle.viewModelScope
import com.hhp227.datemate.data.UserRepository
import com.hhp227.datemate.common.Resource
import kotlinx.coroutines.flow.collectLatest
import kotlinx.coroutines.launch

class SubFirstViewModel(
    private val userRepository: UserRepository,
    savedStateHandle: SavedStateHandle
) : ViewModel() {
    val data: String? = savedStateHandle["data"]

    fun signOut() {
        viewModelScope.launch {
            userRepository.getSignOutResultStream().collectLatest { result ->
                when (result) {
                    is Resource.Error<*> -> {
                    }
                    is Resource.Loading<*> -> {}
                    is Resource.Success<*> -> {}
                }
            }
        }
    }
}